File: e-commerce-frontend/src/types/domains/payment_method.ts
export interface PaymentMethod {
   paymentMethodId: number;
   type: PaymentType;
   providerToken: string;
   last4: string;
   expiryMonth: string;
   expiryYear: string;
   isDefault: boolean;
   cardHolderName: string;
}

export interface PaymentMethodDTO {
   last4: string;
   providerToken: string;
   expiryMonth: string;
   expiryYear: string;
   isDefault: boolean;
   cardHolderName: string;
}

export enum PaymentType {
   CREDIT_CARD = "CREDIT_CARD",
   DEBIT_CARD = "DEBIT_CARD",
   PAYPAL = "PAYPAL",
   BANK_TRANSFER = "BANK_TRANSFER",
   CASH_ON_DELIVERY = "CASH_ON_DELIVERY",
}

File: e-commerce-backend/src/main/java/com/TechKun/model/PaymentMethod.java
package com.TechKun.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@Table(name = "PaymentMethod")
public class PaymentMethod {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "payment_method_id", nullable = false)
    private Integer paymentMethodId;

    @Column(name = "last4", length = 4)
    private String last4 = "4242";

    @Column(name = "provider_token")
    private String providerToken = "stripe";

    @Column(name = "expiry_month", length = 2)
    private String expiryMonth = "12";

    @Column(name = "expiry_year", length = 2)
    private String expiryYear = "28";

    @Column(name = "is_default", nullable = false)
    private Boolean isDefault = false;

    @Column(name = "card_holder_name")
    private String cardHolderName = "NovaCanvas Customer";

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private ShopUser user;
    
    public PaymentMethod(Integer paymentMethodId) {
        this.paymentMethodId = paymentMethodId;
    }
}